src/services/wallets/ArConnectWalletConnector.ts

```diff
@@ -44,4 +53,36 @@ export class ArConnectWalletConnector implements ArweaveWalletConnector {
     );
     return this._ar.winstonToAr(winstonBalance);
   }
+
+  async getWalletANTs(
+    approvedSourceCodeTransactions: string[],
+    cursor?: string,
+  ): Promise<{ ids: string[]; cursor?: string }> {
+    const address = await this.getWalletAddress();
+    const fetchedANTids: Set<string> = new Set();
+    let newCursor: string | undefined = undefined;
+
+    // get contracts deployed by user, filtering with src-codes to only get ANT contracts
+    const deployedResponse = await this._arweave.api.post(
+      '/graphql',
+      deployedAntQuery(address, approvedSourceCodeTransactions, cursor),
+    );
+    if (deployedResponse.data.data?.transactions?.edges?.length) {
+      deployedResponse.data.data.transactions.edges
+        .map((e: any) => ({
+          id: e.node.id,
+          cursor: e.cursor,
+        }))
+        .forEach((ant: { id: string; cursor: string }) => {
+          fetchedANTids.add(ant.id);
+          if (ant.cursor) {
+            newCursor = ant.cursor;
+          }
+        });
+    }
+    return {
+      ids: [...fetchedANTids],
+      cursor: newCursor,
+    };
+  }
```

Comment:
since these queries will likely be the same across all connectors we can create an abstract class called `GraphQLAPI` and separate the interfaces.

```
export interface ArweaveWalletConnector {
  connect(): Promise<void>;
  disconnect(): Promise<void>;
  getWalletAddress(): Promise<string>;
  getWalletBalanceAR(): Promise<string>;
}

export interface ArweaveGraphQLAPI {
  getWalletANTs(
    approvedSourceCodeTransactions: string[],
    cursor?: string,
  ): Promise<{ ids: string[]; cursor?: string }>;
}
```
and then implement the queries in an abstract class 

```
abstract class ArweaveGraphQL implements ArweaveGraphQLAPI {
  constructor(arweave: Arweave) {
      this._arweave = arweave
  }
    async getWalletANTs(
    approvedSourceCodeTransactions: string[],
    cursor?: string,
  ): Promise<{ ids: string[]; cursor?: string }> {
    const address = await this.getWalletAddress();
    const fetchedANTids: Set<string> = new Set();
    let newCursor: string | undefined = undefined;

    // get contracts deployed by user, filtering with src-codes to only get ANT contracts
    const deployedResponse = await this._arweave.api.post(
      '/graphql',
      deployedAntQuery(address, approvedSourceCodeTransactions, cursor),
    );
    if (deployedResponse.data.data?.transactions?.edges?.length) {
      deployedResponse.data.data.transactions.edges
        .map((e: any) => ({
          id: e.node.id,
          cursor: e.cursor,
        }))
        .forEach((ant: { id: string; cursor: string }) => {
          fetchedANTids.add(ant.id);
          if (ant.cursor) {
            newCursor = ant.cursor;
          }
        });
    }
    return {
      ids: [...fetchedANTids],
      cursor: newCursor,
    };
  }
```

and then have the `ArConnectWalletConnector` extend  `ArweaveGraphQL`

export class ArConnectWalletConnector implements ArweaveWalletConnector extends ArweaveGraphQL {
  ...attributes
  constructor(arweave: Arweave) {
     super()
  }
 ...
}
```
also note: we pass `arweave` into the constructor, rather than relying on the static client since we will likey give users to options to change gateways.

Reply:
this approach is `composition` - another approach is `inheritance` where we create a super class, that `ArConnectWalletConnector` can extend, but that's not generally favored when trying to achieve code re-use. a good write up here: https://www.geeksforgeeks.org/favoring-composition-over-inheritance-in-java-with-examples/